src/framework/core/graphicalapplication.cpp

```diff
@@ -124,6 +128,37 @@ void GraphicalApplication::terminate()
     m_terminated = true;
 }
 
+#ifdef __EMSCRIPTEN__
+void GraphicalApplication::mainLoop() {
+    if (m_stopping) {
+        emscripten_cancel_main_loop();
+        MAIN_THREAD_EM_ASM({ window.location.reload(); });
+        return;
+    }
+    mainPoll();
+
+    if (!g_window.isVisible()) {
+        stdext::millisleep(10);
+        return;
+    }
+
+    const auto& FPS = [this] {
+        m_mapProcessFrameCounter.setTargetFps(g_window.vsyncEnabled() || getMaxFps() || getTargetFps() ? 500u : 0u);
+        return m_graphicFrameCounter.getFps();
+    };
+
+    g_drawPool.draw();
+    // update screen pixels
+    g_window.swapBuffers();
+
+    if (m_graphicFrameCounter.update()) {
+        g_dispatcher.addEvent([this, fps = FPS()] {
+            g_lua.callGlobalField("g_app", "onFps", fps);
+        });
+    }
```

Comment:
```suggestion
    if (m_graphicFrameCounter.update()) {
        m_mapProcessFrameCounter.setTargetFps(
            (g_window.vsyncEnabled() || getMaxFps() || getTargetFps()) ? 500u : 0u
        );
        g_dispatcher.addEvent([this, fps = m_graphicFrameCounter.getFps()] {
            g_lua.callGlobalField("g_app", "onFps", fps);
        });
    }
```
then you can remove the entire FPS lambda entirely

Reply:
targetFPS needs to be set before update(), if you look inside update() method, the calculation is based on targetFPS.

https://github.com/mehah/otclient/blob/8d2c0651dd9cefa96728d7cfd2c95a5108dcafd8/src/framework/core/adaptativeframecounter.cpp#L30